chain/ethereum/src/network.rs

```diff
@@ -23,12 +23,26 @@ pub struct EthereumNetworkAdapter {
     limit: usize,
 }
 
-#[derive(Clone)]
+impl EthereumNetworkAdapter {
+    fn is_call_only(&self) -> bool {
+        self.adapter.is_call_only()
+    }
+}
+
+#[derive(Clone, Default)]
 pub struct EthereumNetworkAdapters {
     pub adapters: Vec<EthereumNetworkAdapter>,
+    pub call_only_adapters: Vec<EthereumNetworkAdapter>,
 }
```

Comment:
Would it make sense to have a separate `EthereumCallAdapter` that can only be used to make calls? The way things are right now, you could get a call-only adapter and then use all kinds of different methods on it even though they would blow up. It would be safer to split that out from `EthereumNetworkAdapter`, but I am not sure how much work that will be.

Reply:
I have added some assertions, I thought about this but this would require a significant refactor because we would need to make these functions 2 traits and implement them for different types (normal and call-only) and then change everything that uses these. I do think we should do it, but probably afterwards so that we can start testing ASAP. I think I will add this as a separate issue.